app/controllers/ChoiceController.scala

```diff
@@ -47,14 +47,19 @@ class ChoiceController @Inject()(
 
   private val logger = Logger(this.getClass)
 
-  def displayPage(): Action[AnyContent] = authenticate.async { implicit request =>
+  def displayPage(previousChoice: Option[String] = None): Action[AnyContent] = authenticate.async { implicit request =>
+    def pageForPreviousChoice(previousChoice: Option[String]) = previousChoice match {
+      case Some(choice) => choicePage(Choice.form().fill(Choice(choice)))
```

Comment:
what do you think about
```
val form = Choice.form()
previousChoice.fold(form)(form.fill)
```
?

Reply:
Not sure it adds to readability.  But because I don't do much folding I've done the change